core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -184,21 +189,23 @@ trait EntityComponent {
     }
 
     def entityAttributes(entityId: Rep[Long]): EntityAttributeQuery = for {
-      entityAttrRec <- entityAttributeQuery if entityAttrRec.ownerId === entityId
+      entityAttrRec <- entityAttributeQuery if entityAttrRec.ownerId === entityId && ! entityAttrRec.deleted
     } yield entityAttrRec
 
+    // includes "deleted" hidden entities when using their hidden names
     def findEntityByName(workspaceId: UUID, entityType: String, entityName: String): EntityQuery = {
       filter(entRec => entRec.name === entityName && entRec.entityType === entityType && entRec.workspaceId === workspaceId)
     }
 
     def findEntityByType(workspaceId: UUID, entityType: String): EntityQuery = {
-      filter(entRec => entRec.entityType === entityType && entRec.workspaceId === workspaceId)
+      filter(entRec => entRec.entityType === entityType && entRec.workspaceId === workspaceId && ! entRec.deleted)
     }
 
     def findEntityByWorkspace(workspaceId: UUID): EntityQuery = {
-      filter(_.workspaceId === workspaceId)
+      filter(entRec => entRec.workspaceId === workspaceId && ! entRec.deleted)
```

Comment:
I agree that this is necessary, but I'd prefer to do that as part of a bigger EntityComponent refactoring: https://broadinstitute.atlassian.net/browse/GAWB-1768